mazerunner/src/main/java/com/bugsnag/mazerunner/TestCaseRunnner.java

```diff
@@ -0,0 +1,68 @@
+package com.bugsnag.mazerunner;
+
+import com.bugsnag.Bugsnag;
+import org.apache.log4j.Logger;
+import org.springframework.beans.BeansException;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.ExitCodeGenerator;
+import org.springframework.boot.SpringApplication;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.stereotype.Component;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+@Component
+public class TestCaseRunnner implements CommandLineRunner, ApplicationContextAware {
+
+    private static final Logger LOGGER = Logger.getLogger(TestCaseRunnner.class);
+
+    private ApplicationContext ctx;
+
+    @Override
+    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+        ctx = applicationContext;
+    }
+
+    @Override
+    public void run(String... args) throws Exception {
+        // Put args into the system property so that they can be used later
+        for (String arg : args) {
+            String[] argParts = arg.split("=");
+            LOGGER.info("Setting property " + argParts[0] + "=" + argParts[1]);
```

Comment:
it'd be good to check that there are only 2 parts here, and throw a runtimeexception if that's not the case

Reply:
added check